Sources/Collections/ObservableArray.swift

```diff
@@ -253,6 +253,10 @@ extension ObservableArray: DataSourceProtocol {
   public func numberOfItems(inSection section: Int) -> Int {
     return count
   }
+
+  public func item<Item>(at index: Int) -> Item {
```

Comment:
This all sounds good to me, however I'm having a bit of a conceptual issue. Given:

```swift
extension ObservableArray: QueryableDataSourceProtocol {
  public typealias Item = ???
  public typealias Index = Int

  public var numberOfSections: Int {
    return 1
  }

  public func numberOfItems(inSection section: Int) -> Int {
    return count
  }

  public func item<Item>(at index: Int) -> Item {
    return self[index] as! Item
  }
}
```

What do I type alias `Item` to?